src/elements/content-picker/ContentPicker.js

```diff
@@ -974,7 +972,7 @@ class ContentPicker extends Component<Props, State> {
         }
 
         const { can_set_share_access }: BoxItemPermission = permissions;
-        if (!can_set_share_access) {
+        if (access !== undefined && !can_set_share_access) {
```

Comment:
the `can_set_share_access` permission is only checked if there is a specified access different than the enterprise default